File: Prashant-PB002/wikisearch.rb
#! /usr/bin/env ruby

require 'slop'
require 'wikipedia'

opts = Slop.parse do |o|
  o.string '-s', '--search', 'search term', default: ''
  o.string '--log-file', 'log results to this file (default: wiki.log)', default: 'wiki.log'
  o.bool '-h', '--help', 'show this help'
end

if opts.help?
  puts opts
  exit
end

log_file =  opts[:log_file]

if opts[:search].empty?
  print "Enter search term: "
  search_term = STDIN.gets.chomp
else
  search_term = opts[:search]
end

page = Wikipedia.find(search_term)

# Page does not exist if == '-1'
if page.raw_data['query']['pages'].first[0] == '-1'
  STDERR.puts "No wikipedia article found for '#{search_term}'"
else
  puts page.fullurl
  File.open(log_file, 'a') { |f| f.puts page.fullurl }
end

File: Prashant-PB002/wikisearch.py
#! /usr/bin/env python

import argparse
import sys

from wikiapi import WikiApi

def get_url(query, log_file):
  wiki = WikiApi()
  results = wiki.find(query)
  if len(results) == 0:
    sys.stderr.write("No wikipedia article found for '" + query + "'\n")
  else:
    article = wiki.get_article(results[0])
    print article.url
    with open(log_file, 'a') as f:
      f.write(article.url + "\n")

parser = argparse.ArgumentParser(
  description='Fetch link to wikipedia article for search term'
)

parser.add_argument(
  '--search',
  metavar='TERM',
  type=str,
  nargs='+',
  help="Fetch link to wikipedia article of TERM"
)

parser.add_argument(
  '--log-file',
  metavar='LOG_FILE',
  type=str,
  nargs=1,
  help="Log fetched links to LOG_FILE. By default logs to wiki.log"
)

args = parser.parse_args()

if args.search is None:
  query = raw_input("Enter search term: ")
else:
  query = " ".join(args.search)

if args.log_file is None:
  log_file = 'wiki.log'
else:
  log_file = args.log_file[0]

get_url(query, log_file)
